tests/util/test_misc.py

```diff
@@ -428,3 +431,24 @@ async def test_recording_web_server_specified_response(
         ) as response:
             response.raise_for_status()
             assert await response.json() == expected_response
+
+
+@pytest.mark.parametrize(
+    "obj, type_hint, expected_result",
+    [
+        (42, int, True),
+        (42, uint64, False),
+        (uint64(42), uint64, True),
+        ("42", int, False),
+        ([4, 2], List[int], True),
+        ([4, "2"], List[int], False),
+        ((4, 2), Tuple[int, int], True),
+        ((4, "2"), Tuple[int, int], False),
+        ((4, 2), Tuple[int, ...], True),
+        ((4, "2"), Tuple[int, ...], False),
+        ([(4, Program.to([2]))], List[Tuple[int, Program]], True),
+        ([(4, "2")], Tuple[int, str], False),
```

Comment:
I don't see `Dict` here, even though it looks like `satisfies_hint()` supports it

Reply:
Yep, I added cases that are similar to how we're currently using it, but I can add more if needed.